### ISSUE TITLE
tpm2_getcap prints properties as TPM2_PT_* instead of TPM_PT_* in 3.X branch

### ISSUE THREAD

**Issue by vilemmarsik** [tpm2-software/tpm2-tools#1175]
For my TPM2 device, identified in dmesg as "tpm_tis MSFT0101:00: 2.0 TPM (device-id 0xFE, rev-id 2)", kernel 3.10.0-944.el7.x86_64, and tpm2-tools-3.0.4-1.el7.x86_64, tpm2_getcap prints "TPM_PT_..." instead of "TPM2_PT_...", e.g.:
TPM_PT_HR_NV_INDEX:          0x0000000c
TPM_PT_HR_LOADED:            0x00000000
...
Regular expressions  in test/system fail on this.
Adding a patch to make things work both with "TPM2_*" and "TPM_*".

**Comment by vilemmarsik** [comment]
My patch:
```
diff --git a/test/system/test_tpm2_dictionarylockout.sh b/test/system/test_tpm2_dictionarylockout.sh
index a90854a..5a1f46e 100755
--- a/test/system/test_tpm2_dictionarylockout.sh
+++ b/test/system/test_tpm2_dictionarylockout.sh
@@ -40,16 +40,16 @@ trap onerror ERR
 
 tpm2_dictionarylockout -s -n 5 -t 6 -l 7
 
-if [ "$(tpm2_getcap -c properties-variable | grep TPM2_PT_MAX_AUTH_FAIL | sed -e 's/TPM2_PT_MAX_AUTH_FAIL: \+//')" != "0x00000005" ];then
+if [ "$(tpm2_getcap -c properties-variable | grep TPM2\\?_PT_MAX_AUTH_FAIL | sed -e 's/TPM2\?_PT_MAX_AUTH_FAIL: \+//')" != "0x00000005" ];then
  echo "Failure: setting up the number of allowed tries in the lockout parameters"
  exit 1
 fi
 
-if [ "$(tpm2_getcap -c properties-variable | grep TPM2_PT_LOCKOUT_INTERVAL | sed -e 's/TPM2_PT_LOCKOUT_INTERVAL: \+//')" != "0x00000006" ];then
+if [ "$(tpm2_getcap -c properties-variable | grep TPM2\\?_PT_LOCKOUT_INTERVAL | sed -e 's/TPM2\?_PT_LOCKOUT_INTERVAL: \+//')" != "0x00000006" ];then
  echo "Failure: setting up the lockout period in the lockout parameters"
 fi
 
-if [ "$(tpm2_getcap -c properties-variable | grep TPM2_PT_LOCKOUT_RECOVERY | sed -e 's/TPM2_PT_LOCKOUT_RECOVERY: \+//')" != "0x00000007" ];then
+if [ "$(tpm2_getcap -c properties-variable | grep TPM2\\?_PT_LOCKOUT_RECOVERY | sed -e 's/TPM2\?_PT_LOCKOUT_RECOVERY: \+//')" != "0x00000007" ];then
  echo "Failure: setting up the lockout recovery period in the lockout parameters"
 fi
 
diff --git a/test/system/test_tpm2_nv.sh b/test/system/test_tpm2_nv.sh
index 3b486cc..7d4cbfe 100755
--- a/test/system/test_tpm2_nv.sh
+++ b/test/system/test_tpm2_nv.sh
@@ -153,7 +153,7 @@ tpm2_nvrelease -Q -x 0x1500016 -a 0x40000001
 #
 # Test large writes
 #
-large_file_size=$(tpm2_getcap -c properties-fixed | grep TPM2_PT_NV_INDEX_MAX | sed -r -e 's/.*(0x[0-9a-f]+)/\1/g')
+large_file_size=$(tpm2_getcap -c properties-fixed | grep TPM2\\?_PT_NV_INDEX_MAX | sed -r -e 's/.*(0x[0-9a-f]+)/\1/g')
 nv_test_index=0x1000000
 
 # Create an nv space with attributes 1010 = TPMA_NV_PPWRITE and TPMA_NV_AUTHWRITE
diff --git a/test/system/test_tpm2_quote.sh b/test/system/test_tpm2_quote.sh
index 3c0a9af..3f00d09 100755
--- a/test/system/test_tpm2_quote.sh
+++ b/test/system/test_tpm2_quote.sh
@@ -51,7 +51,7 @@ Handle_ak_quote=0x81010016
 Handle_ek_quote=0x81010017
 Handle_ak_quote2=0x81010018
 
-maxdigest=$(tpm2_getcap -c properties-fixed | grep TPM2_PT_MAX_DIGEST | sed -r -e 's/.*(0x[0-9a-f]+)/\1/g')
+maxdigest=$(tpm2_getcap -c properties-fixed | grep TPM2\\?_PT_MAX_DIGEST | sed -r -e 's/.*(0x[0-9a-f]+)/\1/g')
 if ! [[ "$maxdigest" =~ ^(0x)*[0-9]+$ ]] ; then
  echo "error: not a number, got: \"$maxdigest\"" >&2
  exit 1

```

**Comment by martinezjavier** [comment]
@vilemmarsik I think you found a regression in the 3.X branch.

The tpm2_getcap output doesn't comes from the TPM chip but instead is printed by the tool. But commit ("ac9a5d787753 update to TSS version 2.0") replaced all the `TPM_PT_*` occurrences by `TPM2_PT_*` which is a non-backward compatible change. I'll change the issue subject to better reflect that.

Having said that, I think you should change the tests from the `3.0.X` branch if you are using tpm2-tools version `3.0.5` and not the ones from the `3.X` or master branches.

**Comment by martinezjavier** [comment]
This has been fixed by commit 3598de590a7, it's just that Github only closes it automatically for commits that land in the master branch.
